src/script/squirrel_helper.hpp

```diff
@@ -36,8 +34,6 @@ namespace SQConvert {
 	template <> struct Return<int16_t>        { static inline int Set(HSQUIRRELVM vm, int16_t res)       { sq_pushinteger(vm, res); return 1; } };
 	template <> struct Return<int32_t>        { static inline int Set(HSQUIRRELVM vm, int32_t res)       { sq_pushinteger(vm, res); return 1; } };
 	template <> struct Return<int64_t>        { static inline int Set(HSQUIRRELVM vm, int64_t res)       { sq_pushinteger(vm, res); return 1; } };
-	template <> struct Return<Money>        { static inline int Set(HSQUIRRELVM vm, Money res)       { sq_pushinteger(vm, res); return 1; } };
-	template <> struct Return<TileIndex>    { static inline int Set(HSQUIRRELVM vm, TileIndex res)   { sq_pushinteger(vm, (int32_t)res.base()); return 1; } };
```

Comment:
Need to check this. TileIndex is uint32_t, but for historic reasons I guess, returned as int32_t. ScriptTile's TILE_INVALID remains -1, but things getting returned from the API might not match that any more.

Reply:
Yeah, it fails regression test, forgot to check this :S

Reply:
Any unsigned value has to be converted to signed (squirrel only has SQInteger which is signed).

Reply:
Looking at IDs that might be used by the scripts that are 32 bits:
* SpriteGroupID, ObjectID -> INVALID_ID > 2**31, not in API though
* CargoPacketID, CargoPaymentID, OrderID, PersistentStorageID -> INVALID_ID < 2**31 and not in API
* ClientID, though INVALID_CLIENT_ID = 0

So, TileIndex really seems to be the only one with a potential issue.

I would suggest leaving the return overload for TileIndex as it is, so the behaviour and value remains as it is while not burdening any future strong types to use negative numbers in scripts.